stonesoup/models/measurement/nonlinear.py

```diff
@@ -942,9 +942,9 @@ def inverse_function(self, detection, **kwargs) -> StateVector:
         x, y, z = sphere2cart(rho, phi, theta)
         # because only rho_rate is known, only the components in
         # x,y and z of the range rate can be found.
-        x_rate = np.cos(phi) * np.cos(theta) * rho_rate
-        y_rate = np.cos(phi) * np.sin(theta) * rho_rate
-        z_rate = np.sin(phi) * rho_rate
+        x_rate = np.cos(theta) * np.cos(phi) * rho_rate
+        y_rate = np.cos(theta) * np.sin(phi) * rho_rate
+        z_rate = np.sin(theta) * rho_rate
```

Comment:
```suggestion
        x_rate, y_rate, z_rate = sphere2cart(rho_rate, phi, theta)
```
This should perform the same calculation